tests/admin.test.ts

```diff
@@ -182,3 +185,195 @@ describe('bonus points submission', () => {
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: users[0].email, accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: users[1].email, accessType: UserAccessType.MARKETING },
+          { user: users[2].email, accessType: UserAccessType.MERCH_STORE_DISTRIBUTOR },
+          { user: users[3].email, accessType: UserAccessType.STAFF },
+        ],
+      }, standard);
+    }).rejects.toThrow(ForbiddenError);
+  });
+
+  test('attempt to update duplicate users', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const users = UserFactory.create(5);
+
+    users[0].accessType = UserAccessType.STAFF;
+    users[0].email = 'smhariha@ucsd.edu';
+
+    await new PortalState()
+      .createUsers(users[0], admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: 'smhariha@ucsd.edu', accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: 'smhariha@ucsd.edu', accessType: UserAccessType.STAFF },
+        ],
+      }, admin);
+    }).rejects.toThrow(BadRequestError);
+  });
+
+  // attempt to promote a user to an admin
+  test('attempt to promote a user to an admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const users = UserFactory.create(5);
+
+    users[0].accessType = UserAccessType.STAFF;
+    users[1].accessType = UserAccessType.STANDARD;
+    users[2].accessType = UserAccessType.MARKETING;
+    users[3].accessType = UserAccessType.MERCH_STORE_DISTRIBUTOR;
+
+    await new PortalState()
+      .createUsers(users[0], users[1], users[2], users[3], admin)
+      .write();
+
+    const adminController = ControllerFactory.admin(conn);
+
+    await expect(async () => {
+      await adminController.updateUserAccessLevel({
+        accessUpdates: [
+          { user: users[0].email, accessType: UserAccessType.MERCH_STORE_MANAGER },
+          { user: users[1].email, accessType: UserAccessType.ADMIN },
+          { user: users[2].email, accessType: UserAccessType.MERCH_STORE_DISTRIBUTOR },
+          { user: users[3].email, accessType: UserAccessType.STAFF },
+        ],
+      }, admin);
+    }).rejects.toThrow(ForbiddenError);
+  });
+
+  // attempt to demote an existing admin
+  test('attempt to demote an existing admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    // create another admin user
+    const secondAdmin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+
+    secondAdmin.email = 'test23432@ucsd.edu';
```

Comment:
u can put email into the curly brackets in fake() if there is no point of changing it separately